src/celeritas/decay/interactor/MuDecayInteractor.hh

```diff
@@ -0,0 +1,242 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/decay/interactor/MuDecayInteractor.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Constants.hh"
+#include "corecel/Macros.hh"
+#include "corecel/Types.hh"
+#include "corecel/data/StackAllocator.hh"
+#include "corecel/math/Algorithms.hh"
+#include "celeritas/Quantities.hh"
+#include "celeritas/decay/data/MuDecayData.hh"
+#include "celeritas/phys/FourVector.hh"
+#include "celeritas/phys/Interaction.hh"
+#include "celeritas/phys/ParticleTrackView.hh"
+#include "celeritas/phys/Secondary.hh"
+#include "celeritas/random/distribution/IsotropicDistribution.hh"
+#include "celeritas/random/distribution/RejectionSampler.hh"
+#include "celeritas/random/distribution/UniformRealDistribution.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Perform muon decay.
+ *
+ * Only one decay channel is implemented, with muons decaying to
+ * \f[
+ * \mu^- \longrightarrow e^- \overline{\nu}_e \nu_\mu
+ * \f]
+ * or
+ * \f[
+ * \mu^+ \longrightarrow e^+ \nu_e \overline{\nu}_\mu
+ * \f].
+ *
+ * This interactor follows \c G4MuonDecayChannel and the Physics Reference
+ * Manual, Release 11.2, section 4.2.3.
+ *
+ * \warning Neutrinos are currently not returned in this interactor as they
+ * are not tracked down or transported.
+ *
+ * \note The full three-body decay can be recovered from PR #1456, commit
+ * `ecc4326`.
+ */
+class MuDecayInteractor
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using Energy = units::MevEnergy;
+    using MevMomentum = units::MevMomentum;
+    using Mass = units::MevMass;
+    using UniformRealDist = UniformRealDistribution<real_type>;
+    //!@}
+
+    // Construct with shared and state data
+    inline CELER_FUNCTION
+    MuDecayInteractor(MuDecayData const& shared,
+                      ParticleTrackView const& particle,
+                      Real3 const& inc_direction,
+                      StackAllocator<Secondary>& allocate);
+
+    // Sample an interaction with the given RNG
+    template<class Engine>
+    inline CELER_FUNCTION Interaction operator()(Engine& rng);
+
+  private:
+    //// DATA ////
+
+    // Constant data
+    MuDecayData const& shared_;
+    // Incident muon energy
+    Energy const inc_energy_;
+    // Incident muon direction
+    Real3 const& inc_direction_;
+    // Allocate space for secondary particles (electron only)
+    StackAllocator<Secondary>& allocate_;
+    // Define decay channel based on muon or anti-muon primary
+    bool is_muon_;
```

Comment:
This is confusing... maybe `is_antiparticle_` since if true it creates a positron rather than electron?

Reply:
Or since there's only one secondary now, maybe we should just store the secondary ID instead?